Sources/Logging/Strings/PurchaseStrings.swift

```diff
@@ -202,19 +200,19 @@ extension PurchaseStrings: LogMessage {
         case .purchases_synced:
             return "Purchases synced."
 
-        case let .purchasing_product(product):
-            return "Purchasing Product '\(product.productIdentifier)'"
-
-        case let .purchasing_product_from_package(product, package):
-            return "Purchasing Product '\(product.productIdentifier)' from package " +
-            "in Offering '\(package.presentedOfferingContext.offeringIdentifier)'"
-
-        case let .purchasing_product_with_offer(product, discount):
-            return "Purchasing Product '\(product.productIdentifier)' with Offer '\(discount.identifier)'"
-
-        case let .purchasing_product_from_package_with_offer(product, package, discount):
-            return "Purchasing Product '\(product.productIdentifier)' from package in Offering " +
-            "'\(package.presentedOfferingContext.offeringIdentifier)' with Offer '\(discount.identifier)'"
+        case let .purchasing_product(product, package, discount, metadata):
+            var message = "Purchasing Product '\(product.productIdentifier)'"
+            if let package = package {
+                message += " from package in Offering " +
+                "'\(package.presentedOfferingContext.offeringIdentifier)'"
+            }
+            if let discount = discount {
+                message += " with Offer '\(discount.identifier)'"
+            }
+            if let metadata = metadata {
+                message += " with metadata: \(metadata)"
+            }
+            return message
```

Comment:
Took the opportunity to simplify the log message to it can be conditionally composed depending on the optional parameters instead of creating a new message for every combination.